Dockerfile.ci

```diff
@@ -0,0 +1,34 @@
+# syntax=docker/dockerfile:1.9@sha256:fe40cf4e92cd0c467be2cfc30657a680ae2398318afd50b0c80585784c604f28
```

Comment:
you don't need to add this file, it got moved to https://github.com/cilium/cilium/blob/main/cilium-cli/Dockerfile. apologies for the confusion, we are still in the middle of cleaning things up after https://github.com/cilium/design-cfps/pull/9 🧹

Reply:
I'm now even more confused, if the cilium-cli is now in the other repo, shouldn't we build and release the Docker image from there as well? How do you keep the codebases in sync? Because if we build it here, it might not be the same source as in the other repo.

Reply:
right now we vendor the code from cilium/cilium repo. at some point we might start building cilium-cli releases from cilium/cilium repo, but nobody has signed up to drive it.

... and i just realized we still need to keep this Dockerfile for now. sorry for all the confusion, i suggest focusing on getting the optimized Dockerfile reviewed and merged.